src/Cello/problem_Problem.hpp

```diff
@@ -137,6 +137,14 @@ class Problem : public PUP::able
   /// Return the named method object if present
   Method * method (std::string name) const throw();
 
+  // Return whether a method object with given name exists for this problem
+  bool method_exists (std::string name) const throw();
+
+  // Returns true if method objects with both given names exist for this problem,
+  // and the method called "name1" precedes the method called "name2" in the
+  // method list. Returns false otherwise.
+  bool method_precedes(std::string name1, std::string name2) const throw();
```

Comment:
Same advice as above: you might want to consider passing a `const std::string&` instead, but it's not a big deal (since you only really use this while setting things up)